math/src/main/java/com/powsybl/math/graph/GraphUtil.java

```diff
@@ -53,13 +54,14 @@ public ConnectedComponentsComputationResult(int[] componentNumber, int[] ordered
     }
 
     private static void computeConnectedComponents(int v1, int c, int[] componentSize, TIntArrayList[] adjacencyList, int[] componentNumber) {
-        componentNumber[v1] = c;
-        ++componentSize[c];
-        TIntArrayList ls = adjacencyList[v1];
-        for (int i = 0; i < ls.size(); i++) {
-            int v2 = ls.getQuick(i);
-            if (componentNumber[v2] == -1) {
-                computeConnectedComponents(v2, c, componentSize, adjacencyList, componentNumber);
+        LinkedList<Integer> nodes = new LinkedList<>();
```

Comment:
Why not a Queue/Stack ? Wouldn't it be faster for a push/pop data structure?

Reply:
Replaced LinkedList with ArrayDeque